src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java

```diff
@@ -129,79 +138,97 @@ public IBlockState getActualState(final IBlockState state, final IBlockAccess wo
         if (rack.isEmpty() && (rack.getOtherChest() == null || rack.getOtherChest().isEmpty()))
         {
-            if(rack.getOtherChest() != null)
+            if (rack.getOtherChest() != null)
             {
-                if(rack.isMain())
+                if (rack.isMain())
                 {
-                    return state.withProperty(BlockMinecoloniesRack.VARIANT, BlockMinecoloniesRack.EnumType.DEFAULTDOUBLE)
-                            .withProperty(FACING, BlockPosUtil.getFacing(rack.getNeighbor(), pos));
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, RackType.DEFAULTDOUBLE)
+                             .withProperty(FACING, BlockPosUtil.getFacing(rack.getNeighbor(), pos));
                 }
                 else
                 {
-                    return state.withProperty(BlockMinecoloniesRack.VARIANT, EnumType.EMPTYAIR);
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, RackType.EMPTYAIR);
                 }
             }
             else
             {
-                return state.withProperty(BlockMinecoloniesRack.VARIANT, EnumType.DEFAULT);
+                return state.withProperty(BlockMinecoloniesRack.VARIANT, RackType.DEFAULT);
             }
         }
         else
         {
-            if(rack.getOtherChest() != null)
+            if (rack.getOtherChest() != null)
             {
-                if(rack.isMain())
+                if (rack.isMain())
                 {
-                    return state.withProperty(BlockMinecoloniesRack.VARIANT, EnumType.FULLDOUBLE)
-                            .withProperty(FACING, BlockPosUtil.getFacing(rack.getNeighbor(), pos));
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, RackType.FULLDOUBLE)
+                             .withProperty(FACING, BlockPosUtil.getFacing(rack.getNeighbor(), pos));
                 }
                 else
                 {
-                    return state.withProperty(BlockMinecoloniesRack.VARIANT, BlockMinecoloniesRack.EnumType.EMPTYAIR);
+                    return state.withProperty(BlockMinecoloniesRack.VARIANT, RackType.EMPTYAIR);
                 }
             }
             else
             {
-                return state.withProperty(BlockMinecoloniesRack.VARIANT, EnumType.FULL);
+                return state.withProperty(BlockMinecoloniesRack.VARIANT, RackType.FULL);
             }
         }
     }
 
     /**
-     * Check if a certain block should be replaced with a rack.
-     * @param block the block to check.
-     * @return true if so.
+     * Convert the BlockState into the correct metadata value.
+     *
+     * @deprecated (Remove this as soon as minecraft offers anything better).
      */
-    public static boolean shouldBlockBeReplacedWithRack(final Block block)
+    @NotNull
+    @Override
+    @Deprecated
+    public IBlockState withRotation(@NotNull final IBlockState state, final Rotation rot)
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)